layouts/fastly/helix.vcl

```diff
@@ -636,77 +841,43 @@ sub vcl_fetch {
     esi;
   }
 
-  if (beresp.http.X-Static == "Raw/Static") {
-    if (beresp.status == 307) {
-      // static.js returns a 307 if the response is greater than OpenWhisk's
-      // delivery limit. We restart the request and deliver directly from
-      // GitHub instead, carring over the Content-Type header that static.js guessed
-      set req.http.X-Static = "Redirect";
-      set req.url = beresp.http.Location;
-      set req.http.X-Content-Type = beresp.http.X-Content-Type;
-
-      set req.http.X-Trace = req.http.X-Trace + "; RESTART";
-      restart;
-    } else {
-      return(deliver);
-    }
-  } elseif (req.http.X-Static == "Redirect") {
-    // and this is where we fix the headers of the GitHub static response
-    // so that they become digestible by a browser.
-    // - recover Content-Type from X-Content-Type
-    // - filter out GitHub-headers
-
-    set beresp.http.Content-Type = req.http.X-Content-Type;
-    unset beresp.http.X-Content-Type-Options;
-    unset beresp.http.X-Frame-Options;
-    unset beresp.http.X-XSS-Protection;
-    unset beresp.http.Content-Security-Policy;
-
-  } elseif ((beresp.status == 404 || beresp.status == 204) && !req.http.X-Disable-Static) {
-    # That was a miss. Let's try to restart.
-    set beresp.http.X-Status = beresp.status + "-Restart " + req.restarts;
-    set beresp.status = 404;
-
-    if (req.http.X-Static == "Static") {
-      set req.http.X-Static = "Dynamic";
-      set beresp.http.X-Static = "Dynamic";
-    } else {
-      set req.http.X-Static = "Static";
-      set beresp.http.X-Static = "Static";
-    }
-    set req.url = req.http.X-URL;
-
-    set req.http.X-Trace = req.http.X-Trace + "; RESTART";
-    restart;
-  }
-
+  call hlx_fetch_static;
 
   return(deliver);
 }
 
 sub vcl_hit {
-#FASTLY hit
   set req.http.X-Trace = req.http.X-Trace + "; vcl_hit";
+#FASTLY hit
 
   if (!obj.cacheable) {
     return(pass);
   }
   return(deliver);
 }
 
-sub vcl_miss {
-#FASTLY miss
-  set req.http.X-Trace = req.http.X-Trace + "; vcl_miss";
-  unset bereq.http.X-Orig-Url;
+/**
+ * Do all BackEnd REQuest changes. One sub to be called from both vcl_miss and
+ * vcl_pass, to avoid code-rot through having to update two places.
+ */
+sub hlx_bereq {
+  set req.http.X-Trace = req.http.X-Trace + "; hlx_bereq";
+
+  # If we're going to a shield (another Fastly POP) use the original URL and
+  # Host header. If not a shield, we're going to origin; set the URL and Host
+  # header as explained at the top of this file.
   if (req.backend.is_shield) {
     set bereq.url = req.http.X-Orig-Url;
-  }
-
-  # set backend host
-  if (req.backend == F_AdobeRuntime) {
-    set bereq.http.host = "adobeioruntime.net";
-  } elsif (req.backend == F_GitHub) {
-    set bereq.http.host = "raw.githubusercontent.com";
+    set bereq.http.Host = req.http.X-Orig-Host;
```

Comment:
we stash the original host in `req.http.X-Orig-Host`, however, we never actually change `req.http.host`
Is this risk management against future changes to host in vcl_recv, or just an oversight/relic from an older version.

Reply:
Risk management vs future changes